internal/refactoring/cross_provider_move.go

```diff
@@ -90,6 +90,13 @@ func (m *crossTypeMover) prepareCrossTypeMove(stmt *MoveStatement, source, targe
 		return nil, diags
 	}
 
+	sourceProvider, err := m.getProvider(sourceProviderAddr.Provider)
```

Comment:
You can't rely on a source provider in a `MoveResourceState` call, because we may not have the provider at all any longer. This is why the `SourceState` is the raw json, because we cannot decode it, and it's up to the provider to interpret the data.

Reply:
Sorry, I was thinking of a different workflow on the way out yesterday, you are correct that in this case Terraform will (at least currently) enforce the existence of the provider. What it won't do is ensure the provider schema matches, because it's not going through any state upgrade process, so we can't be sure that the provider can decode the state without an upgrade call first. Aside from that though, there's no reason to decode the value, because you are just turning around and re-encoding back to the wire format. It's always going to be up to the receiving provider to do the final decoding of the values.

Reply:
Yeah good point, I got rid of the de and re-encoding, I think this makes more sense now 👍